### DISCUSSION TITLE
Why are Ref given as a lambda ?

### DISCUSSION THREAD

**Discussion by luzzardik** [typegoose/typegoose#916] [Q&A] [answered]
Hi ! I'm trying to understand why the docs asks us to give the refs as such `@prop({ref : () => MyRef})`, is there any reason ?
Couldn't we just use `@prop({ref : MyRef})` ?

Thanks !

**Comment by hasezoey** [comment, marked as answer]
Please read the documentation [Reference other Classes: Common Problems](https://typegoose.github.io/typegoose/docs/guides/advanced/reference-other-classes/#common-problems)

small snippits:
>  If `() =>` is not used, there can be problems when the class (/ variable) is defined after the decorator that requires it.

> Because of the order classes are loaded and reordered at runtime, this might result in some references being null / undefined / not existing.

> The new `() => Class` is meant to help with [Circular Dependencies](https://typegoose.github.io/typegoose/docs/guides/advanced/reference-other-classes#circular-dependencies), but cannot remove the problems in all cases, see Circular Dependencies for more.
